controller/k8s/api.go

```diff
@@ -174,6 +174,7 @@ func NewAPI(k8sClient kubernetes.Interface, spClient spclient.Interface, resourc
 
 // Sync waits for all informers to be synced.
 func (api *API) Sync() {
```

Comment:
since we're calling `Sync()` at non-startup times now, consider adding a comment that this method may kill the process if sync fails. we could go a step further and have it return an error rather than exit, but maybe not worth it. also consider adding a log line (warning? info?) to help us better understand how often this is happening.

(un)related: is it safe to call this method concurrently?

Reply:
Re concurrency safety, the underlying functions there are the `HasSynced` functions, which at least are locked per sharedInformer:

https://github.com/kubernetes/client-go/blob/7de88b14dcc8f8ebdc2953e776666387205cb592/tools/cache/shared_informer.go#L283-L291